### PR TITLE
Upgrade Engine Android SDK to 35

### PR DESCRIPTION
**Pull request by camsim99**
**REVIEW BUT DO NOT MERGE:** Will merge after https://github.com/flutter/buildroot/pull/870 lands.

1: Updates engine to use Android 35. Part of https://github.com/flutter/flutter/issues/150215.
Engine step of https://github.com/flutter/flutter/blob/master/docs/platforms/android/Upgrading-Engine's-Android-API-version.md.

2: Updates test runner AGP version to 8.5, engine Gradle version to 8.9 to accommodate update.

3: Fixes lint and suppresses deprecation warnings caused by the update. None of the deprecations are unknown by the team (they are handled in the code), but the tests emitted warnings.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by camsim99** [comment]
Looks like I need to update the Gradle version to 8.4 when I get the chance. Use https://github.com/flutter/engine/blob/b369582a94f1204f51bc913eeaa30f250602501e/tools/gradle/README.md?plain=1#L20

**Comment by camsim99** [comment]
I also need to bump the targetSDK versions

**Comment by camsim99** [comment]
TODO: udpate packages.txt

**Comment by camsim99** [comment]
Failures seem unrelated? Submitting for review

**Review by gmackall — APPROVED** [review]
Changes lgtm (as long as the pointing to fork for test purpose is removed) - total guess but are these test failures related to the buildmoot effort?

```
Creating file ./src/flutter/GeneratedPluginRegistrant.java
PASS: ban_generated_plugin_registrant_java failed as expected
Creating file ./src/third_party/GeneratedPluginRegistrant.java
touch: cannot touch '/b/s/w/ir/cache/builder/src/third_party/GeneratedPluginRegistrant.java': No such file or directory
```

I believe at some point recently the `src/third_party` directory had reached uselessness (everything migrated out), perhaps the test needs to be changed to not try to create a file in that directory?

https://github.com/flutter/engine/blob/312b51988abd56859699208303180bb4d2bd2511/ci/test/ban_generated_plugin_registrant_java_test.sh#L63

cc @matanlurey

File: shell/platform/android/test_runner/build.gradle
Comment (reidbaker) [inline shell/platform/android/test_runner/build.gradle:7]: can this be 8.5?

**Review by reidbaker — APPROVED** [review]

**Comment by reidbaker** [comment]
Build failure here https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8742043312500311153/+/u/test:_ban_test_GeneratedPluginRegistrant.java/stdout 
I think the third_party directory moved.

File: shell/platform/android/test_runner/build.gradle
Comment (camsim99) [reply shell/platform/android/test_runner/build.gradle]: Done

**Comment by camsim99** [comment]
> I believe at some point recently the src/third_party directory had reached uselessness (everything migrated out), perhaps the test needs to be changed to not try to create a file in that directory?

Removed this test case because I believe it was just testing for accidentally putting that GeneratedPluginRegistrant file in the third_party directory, which should not be possible anymore based on that test failure/buildmoot stuff? Correct me if I'm wrong.

**Comment by reidbaker** [comment]
> > I believe at some point recently the src/third_party directory had reached uselessness (everything migrated out), perhaps the test needs to be changed to not try to create a file in that directory?
> 
> Removed this test case because I believe it was just testing for accidentally putting that GeneratedPluginRegistrant file in the third_party directory, which should not be possible anymore based on that test failure/buildmoot stuff? Correct me if I'm wrong.

It is still valuable but the relative location of the directory changed. I think this should be an updated not removed test.

**Comment by camsim99** [comment]
Landing this + buildroot changed blocked by https://github.com/flutter/flutter/issues/152177.

**Review by reidbaker — APPROVED** [review]

**Pull request merged**
